### PR TITLE
Fix/delayofflinewarning

### LINKED ISSUE #39250: Delay offline warning?
Would it be better to only show users the offline warning if they've been offline for a few seconds?  Since we retry failed submissions, they should be able to work successfully even if their internet is patchy.

### PR DESCRIPTION
**Pull request by ruhci28**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [ x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ x] My pull request targets the `master` branch of freeCodeCamp.
- [ x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #39250 

I delayed the offline warning for 5 seconds when the user is offline. The warning will not be shown if in the meantime the user get their connection back with the internet .  The warning will disappear after the user come online.

### CODE REVIEW

File: curriculum/challenges/english/03-front-end-libraries/react/add-event-listeners.english.md [hunk 1]
```diff
@@ -62,8 +62,9 @@ class MyComponent extends React.Component {
   // change code above this line
   handleEnter() {
     this.setState({
-      message: this.state.message + 'You pressed the enter key! '
-    });
+    message: this.state.message + 'You pressed the enter key! '
+  }
+      );
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/03-front-end-libraries/react/add-event-listeners.english.md:67]:
```suggestion
      message: this.state.message + 'You pressed the enter key! '
    });
```
This should revert the changes.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
It's looking good, but could you revert the changes to the challenge?

File: curriculum/challenges/english/03-front-end-libraries/react/add-event-listeners.english.md (hunk 1 above)
Comment (ruhci28) [reply curriculum/challenges/english/03-front-end-libraries/react/add-event-listeners.english.md:67]: yeah sure

**Review by ojeytonwilliams — DISMISSED** [review]
Great work, @ruhci28!  Looks good to me :+1:

**Review by naomi-lgbt — DISMISSED** [review]
LGTM - gave this a test on a local instance. Simulating offline-status with dev tools correctly generated the warning after a few seconds. 🙂

File: client/src/components/OfflineWarning/OfflineWarning.js [hunk 2]
```diff
@@ -3,17 +3,32 @@ import PropTypes from 'prop-types';
 
 import './offline-warning.css';
 
+const delayInMilliSeconds = 5000;
+var id;
```
Comment (RandellDawson) [inline client/src/components/OfflineWarning/OfflineWarning.js:7]: Can we use `let` here`?
```suggestion
let id;
```

Comment (ruhci28) [reply client/src/components/OfflineWarning/OfflineWarning.js:7]: Yeah

Comment (ruhci28) [reply client/src/components/OfflineWarning/OfflineWarning.js]: @RandellDawson i apply the changes

**Review by ojeytonwilliams — APPROVED** [review]
@RandellDawson it's ready to merge when you're happy with it.

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
Congrats @ruhci28 on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Comment by ruhci28** [comment]
Thank you . A big thanks to @ojeytonwilliams  for guiding me .

**Comment by ojeytonwilliams** [comment]
You're very welcome, @ruhci28.  Thanks for helping improve freeCodeCamp - it's much appreciated.

**Pull request merged**